//Protocol buffers for project

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.dist_sgd";
option java_outer_classname = "DistSGD";
option objc_class_prefix = "DSG";

package dist_sgd;

//
service ParamFeeder {
  // Sends the parameters back and forth between server and client
  rpc SendParams (Params) returns (Params) {}
}

// A very simple tensor
message Params {
  // The number of parameters
  int32 tensor_len = 1;

  int32 client_id = 2;

  int32 data_indx = 3;

  // DT_FLOAT.
  repeated float float_val = 4 [packed = true];
}




